examples/llm/tests/test_model.py

```diff
@@ -336,3 +339,244 @@ def test_opt_wrapping(prefixlm):
+        pytest.skip(
+            f'This test requires CUDA to be available in order to run with {attention_impl} attention.'
+        )
+
+    reproducibility.seed_all(1234)
+    device = get_device(device)
+
+    hf_config = MosaicGPTConfig(
+        init_device='cpu',
+        d_model=128,
+        n_heads=4,
+        n_layers=2,
+        mlp_ratio=2,
+        max_seq_len=2048,
+        emb_pdrop=0.1,
+        resid_pdrop=0.2,
+        attn_impl=attention_impl,
+    )
+    mosaic_gpt = MosaicGPT(hf_config)
+    mosaic_gpt.eval()
+    mosaic_gpt = device.module_to_device(mosaic_gpt)
+
+    left_padding_input_ids = torch.tensor(
+        [[50256, 50256, 50256, 11274, 16390, 11],
+         [50256, 50256, 50256, 11274, 16390, 11]])
+    left_padding_input_ids = device.tensor_to_device(left_padding_input_ids)
+    left_padding_attention_mask = torch.tensor([[0, 0, 0, 1, 1, 1],
+                                                [0, 0, 0, 1, 1, 1]])
+    left_padding_attention_mask = device.tensor_to_device(
+        left_padding_attention_mask)
+
+    no_padding_input_ids = torch.tensor([[11274, 16390, 11], [11274, 16390,
+                                                              11]])
+    no_padding_input_ids = device.tensor_to_device(no_padding_input_ids)
+    no_padding_attention_mask = torch.tensor([[1, 1, 1], [1, 1, 1]])
+    no_padding_attention_mask = device.tensor_to_device(
+        no_padding_attention_mask)
+
+    batched_input_ids = torch.tensor([[50256, 50256, 50256, 11274, 16390, 11],
+                                      [50256, 50256, 16, 11274, 16390, 11]])
+    batched_input_ids = device.tensor_to_device(batched_input_ids)
+    batched_attention_mask = torch.tensor([[0, 0, 0, 1, 1, 1],
+                                           [0, 0, 1, 1, 1, 1]]).bool()
+    batched_attention_mask = device.tensor_to_device(batched_attention_mask)
+
+    with get_precision_context('amp_bf16' if device.name == 'gpu' else 'fp32'):
+        batched_generation = mosaic_gpt.generate(
+            input_ids=batched_input_ids,
+            attention_mask=batched_attention_mask,
+            max_new_tokens=5)
+        assert batched_generation.shape == (2, 6 + 5)
+
+        reproducibility.seed_all(1234)
+        generation_with_left_padding = mosaic_gpt.generate(
+            input_ids=left_padding_input_ids,
+            attention_mask=left_padding_attention_mask,
+            max_new_tokens=5)
+        assert generation_with_left_padding.shape == (2, 6 + 5)
+        reproducibility.seed_all(1234)
+        generation_with_no_padding = mosaic_gpt.generate(
+            input_ids=no_padding_input_ids,
+            attention_mask=no_padding_attention_mask,
+            max_new_tokens=5)
+        assert generation_with_no_padding.shape == (2, 3 + 5)
+
+        assert generation_with_no_padding[:, 3:].equal(
+            generation_with_left_padding[:, 6:])
+
+
+def check_hf_model_equivalence(model1, model2):
```

Comment:
Why not also check if parameters are the same? (torch.equal)

Reply:
good call, copied from a composer test where the params aren't the same, but makes sense to add here.